adi/adar1000.py

```diff
@@ -781,102 +1257,431 @@ class adar1000_array(context_manager):
+
+    """ Public Methods """
+
+    def initialize_devices(self):
+        """ Initialize the ADAR1000s in the array """
+        for device in self.devices.values():
+            device.initialize()
+        return
+
+    def latch_rx_settings(self):
+        """ Latch in new Gain/Phase settings for the Rx """
+        for device in self.devices.values():
+            device.latch_rx_settings()
+        return
+
+    def latch_tx_settings(self):
+        """ Latch in new Gain/Phase settings for the Tx """
+        for device in self.devices.values():
+            device.latch_tx_settings()
+        return
+
+    def steer_rx(self, azimuth, elevation):
+        """Steer the Rx array in a particular direction. This method assumes that the entire array is one analog beam.
+
+        parameters:
+            azimuth: float
+                Desired beam angle in degrees for the horizontal direction.
+            elevation: float
+                Desired beam angle in degrees for the vertical direction.
+        """
+
+        # Calculate the Φ angles for each element in both directions (in degrees)
+        azimuth_phi, elevation_phi = self._calculate_phi(azimuth, elevation)
+
+        # Update the class variables
+        self._rx_azimuth = azimuth
+        self._rx_elevation = elevation
+        self._rx_azimuth_phi = azimuth_phi
+        self._rx_elevation_phi = elevation_phi
+
+        # Steer the elements in the array
+        for element in self.elements.values():
+            # Calculate the row and column phases
+            column_phase = element.column * azimuth_phi
+            row_phase = element.row * elevation_phi
+
+            element.rx_phase = column_phase + row_phase
+
+        # Latch in the new phases
+        self.latch_rx_settings()
+        return
+
+    def steer_tx(self, azimuth, elevation):
+        """Steer the Tx array in a particular direction. This method assumes that the entire array is one analog beam.
+
+        parameters:
+            azimuth: float
+                Desired beam angle in degrees for the horizontal direction.
+            elevation: float
+                Desired beam angle in degrees for the vertical direction.
+        """
+
+        # Calculate the Φ angles for each element in both directions (in degrees)
+        azimuth_phi, elevation_phi = self._calculate_phi(azimuth, elevation)
+
+        # Update the class variables
+        self._tx_azimuth = azimuth
+        self._tx_elevation = elevation
+        self._tx_azimuth_phi = azimuth_phi
+        self._tx_elevation_phi = elevation_phi
+
+        # Steer the elements in the array
+        for element in self.elements.values():
+            # Calculate the row and column phases
+            column_phase = element.column * azimuth_phi
+            row_phase = element.row * elevation_phi
+
+            element.tx_phase = column_phase + row_phase
+
+        # Latch in the new phases
+        self.latch_tx_settings()
+        return
```

Comment:
Remove

Reply:
Why remove all of these here and below? If these aren't there, none of the new values will take effect...

Reply:
What is the purpose of the return statement?

Reply:
I didn't realize you were referring to the return, I thought you were saying to remove the latch command. I get that it doesn't return anything, but it works the same either way. Just a habit. I can remove it.